Automattic/simplenote-android
# Add Posting to WordPress

New feature time! This PR adds support for sharing a note directly to a WordPress post. We use the WordPress.com REST API and OAuth flow to authenticate a WordPress.com user.

<img width="281" alt="screen shot 2018-11-21 at 2 12 54 pm" src="https://user-images.githubusercontent.com/789137/48870946-9b879a00-ed97-11e8-971a-d722a0b26ef9.png">

I've added `okhttp` to handle the API calls. We had previously had an OAuth flow for the WordPress.com sign in feature, so I've abstracted that code into a static class: `WordPressUtils` to fix some DRY. 

**To Test**
* From the note editor, select the share button and then 'Post to WordPress'. If you haven't authorized with WordPress, it should ask you to connect.
* Once connected, ensure your sites are listed properly. Select one and click on `Post` (you can make it a draft post if desired using the checkbox).
* Ensure the post uploaded properly and is formatted as expected with the note title extracted if present and used as the post title.

Overall UX feedback is welcome as well! We don't have a designer so I rolled this all myself from scratch.
